### DISCUSSION TITLE
Desperate - recognition doesn't work

### DISCUSSION THREAD

**Discussion by dominikaherbst** [jomjol/AI-on-the-edge-device#3356] [General] [answered]
Hi,
I am close to giving up.
With some digits the recognition works, but then some digits change and suddenly a 4 is recognized as a 1, a 3 as a 1, a 2 as a 7 ... 
I believe my problem might be that the digits do not nicely center on the meter. Some digits might come to rest in a slightly higher or lower position than their neighbors ... 
I enabled and disabled "Synchronize y, Δx and Δy between ROIs", same with aspect ratio. I change the ROI positions and adjusted them as described. I even checked whether my numbers are already in the models (https://jomjol.github.io/neural-network-digital-counter-readout/) and they are (I think).
![ROIs](https://github.com/user-attachments/assets/b0c15ea0-f52f-40d9-a89b-9ff73e894be5)
![wrong](https://github.com/user-attachments/assets/ba551edf-4afd-4600-aa44-99a78ea04979)

In my opinion the ROI boxes are too large, but to fit the inner box exactly to the number, they have to be that large.
I tried all models, but I don't get any model consistently working. Sometimes a model works with the ROIs, the reference image that I took, and numbers. But then I take a shower, some digits change, the recognition fails, and I am off by 500-1000L ... Sometimes that even happens with numbers that haven't changed.

It took me a couple of weeks to get everything installed without reflection, ordering extra parts, etc., and now I am close to giving up ... 

Can anyone give me some advise?

Thanks!

<h3>Build Info</h3>

Firmware version: | Release: v15.7.0 (Commit: 0d0b018+)
Firmware build time: | 17.02.2024 00:15
Web interface version: | Release: v15.7.0 (Commit: 0d0b018+)

**Comment by SybexX** [comment]
You could try version 16.0.0-RC4 and then use the zoom function to increase the ROI size. For me, the size 136x58 turned out to be the best.
Then you can play around with the camera settings to improve the image.

**Reply by dominikaherbst** [reply]
Thanks for replying!
What do you mean by zoom function? I can't find any option to zoom in the interface.
I upgraded and had to define a new reference image. The new images seem to have a bit more contrast, however, it still doesn't recognize the right number 4-->6, ... .
Thanks!

**Reply by SybexX** [reply, marked as answer]
![Unbenannt](https://github.com/user-attachments/assets/7a366b85-58b9-4900-9094-970bbafb7ae3)

**Reply by dominikaherbst** [reply]
NICE!!
The zoom seems to have improved it a bit. I will test it for a while.
Thank you!

**Reply by dominikaherbst** [reply]
I tested it for a week. It is not perfect yet, but good enough :-)
Thanks for your help! 

For everyone else who is interested, with the zoom it looked like this:
![alg_roi](https://github.com/user-attachments/assets/d5475d06-83e0-4b51-9774-5d474774151f)

Not having the same aspect ratio was also critical for me, because the dials vary in width.
